### PR TITLE
Added Spectral and VolDev Split to Hencky Model

### LINKED ISSUE #84: Add VolDev and Spectral split support for Hencky Isotropic Elasticity
## Reason

It useful to be able to easily run both splits 
## Design

Add split to .C and .h for this type

### PR DESCRIPTION
**Pull request by TheGreatCid**
Added spectral and Voldev split to HeckyIsotropicElasticity
Closes #84

### CODE REVIEW

**Comment by hugary1995** [comment]
Sorry I've been kept with some other things. I swear I'll look at this next week.

**Comment by TheGreatCid** [comment]
Have you had a chance to look at this?
I am not sure why checks are failing

**Comment by hugary1995** [comment]
I was working on #79 to get tests to pass. After #79 is merged, you can rebase this branch and I will merge it.

You can see why the tests are failing by clicking on details. In your case the Pretest failed because the PR doesn't reference a valid issue.

**Comment by hugary1995** [comment]
Sorry for taking so long. I just merged the fix to the testing system. Could you now rebase your branch? I'll review it in the meantime. Thanks!

File: src/materials/large_deformation_models/HenckyIsotropicElasticity.C [hunk 1]
```diff
@@ -95,3 +99,81 @@ HenckyIsotropicElasticity::computeMandelStressNoDecomposition(const ADRankTwoTen
...
   return stress;
 }
+
+ADRankTwoTensor
+HenckyIsotropicElasticity::computeMandelStressVolDevDecomposition(const ADRankTwoTensor & Fe,
+                                                                  const bool plasticity_update)
+{
+  ADRankTwoTensor strain = Fe;
+  // If this is called during a plasticity update, we need to first exponentiate Fe, where Fe should
+  // be some plastic flow. The foolowing operations cancel out with an exponentiation of Fe, so we
```
Comment (hugary1995) [inline src/materials/large_deformation_models/HenckyIsotropicElasticity.C:109]:
```suggestion
  // be some plastic flow. The following operations cancel out with an exponentiation of Fe, so we
```

File: src/materials/large_deformation_models/HenckyIsotropicElasticity.C [hunk 2]
```diff
@@ -95,3 +99,81 @@ HenckyIsotropicElasticity::computeMandelStressNoDecomposition(const ADRankTwoTen
...
+  // only do this in the case of exponentiate == false
```
Comment (hugary1995) [inline src/materials/large_deformation_models/HenckyIsotropicElasticity.C:110]:
```suggestion
  // only do this in the case of plasticity_update == false
```
I know you probably just copied it from some other place written by me. I think I said it wrong there as well.

File: src/materials/large_deformation_models/HenckyIsotropicElasticity.C [hunk 3]
```diff
@@ -95,3 +99,81 @@ HenckyIsotropicElasticity::computeMandelStressNoDecomposition(const ADRankTwoTen
...
+}
+
+ADRankTwoTensor
+HenckyIsotropicElasticity::computeMandelStressSpectralDecomposition(const ADRankTwoTensor & Fe,
+                                                                    const bool plasticity_update)
+{
+  ADRankTwoTensor strain = Fe;
+  // If this is called during a plasticity update, we need to first exponentiate Fe, where Fe should
+  // be some plastic flow. The following operations cancel out with an exponentiation of Fe, so we
+  // only do this in the case of exponentiate == false
```
Comment (hugary1995) [inline src/materials/large_deformation_models/HenckyIsotropicElasticity.C:149]:
```suggestion
  // only do this in the case of plasticity_update == false
```

**Review by hugary1995 — CHANGES_REQUESTED** [review]
Overall looks very good. Thanks for your contribution!

It would be nice if you can add this two new formulations in the HenckyIsotropicElasticity.md documentation as well.

**Comment by TheGreatCid** [comment]
I'll update the documentation tomorrow! I'll let you know when that is ready

**Comment by hugary1995** [comment]
The precheck fails because none of the commit messages contains an issue number. Try to amend your last commit and force push.

**Comment by TheGreatCid** [comment]
Don't merge this quite yet. Still need to update documentation. I just wanted to make sure I could fix my last commit

**Review by hugary1995 — APPROVED** [review]

**Pull request merged**